pkg/service/auth_context.go

```diff
@@ -251,7 +251,7 @@ func (authContext *AuthContext) GetMetadata() map[string]interface{} {
 
 func (authContext *AuthContext) FindIdentityByName(name string) (interface{}, error) {
 	for identityConfig := range authContext.Identity {
-		if identityConfig.OIDC.Name == name {
+		if identityConfig.OIDC != nil && identityConfig.OIDC.Name == name {
```

Comment:
Hmm. I think this nil-check should be done in the client, not here. E.g. https://github.com/3scale-labs/authorino/blob/6a9c800929e7a48d90e6a2a791d26883582ec90c/pkg/config/metadata/user_info.go#L25-L29

Reasons:
1. it is more consistent with the name of the function which suggests finding a config and nothing else.
2. Because the function returns the config, not the evaluated identity, after all.

Reply:
BTW, even fetching only OIDC identity configs here is wrong. That's why I'm proposing to change it in #53.

Reply:
It makes sense, I'll change it. However, there's something that smells, I think that relying on `OIDC` not being `nil` is a bit flawed since it's possible that we don't want `OIDC` as an identity method in the first place (as in this PR case). Maybe the problem relies on how we are storing the `Identity` config, we should find a better way and not being implicit about those methods.

Reply:
Didn't see your second comment until after reloading the page hehe. Yes, that proposing change is more or less what I'm talking about